### DISCUSSION TITLE
Question about Algeb

### DISCUSSION THREAD

**Discussion by jinningwang** [CURENT/andes#175] [Modeling] [answered]
@cuihantao 

Hi Hantao,

I have a question about the `Algeb`. There is a parameter named `discrete`, and the document said: *Will call check_var on the discrete component.* May I ask why do we need this?

Background:
I'm developing `PIDController` and `PIDAWHardLimit`, and I'm referring to the existing `PIAWHardLimit`.
There is a code: `self.y.discrete = self.hl`, I'm not sure what's the purpose of this design.

Regards,
Jinning

**Comment by cuihantao** [comment, marked as answer]
The discrete component is what the algebraic variable _depends on_. The discrete flags will be evaluated before initializing the algebraic variable.

**Reply by jinningwang** [reply]
Thanks for your reply!

Regards,
Jinning
